components/button/src/Button.js

```diff
@@ -0,0 +1,276 @@
+/* @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { mediaQueries } from './utils';
```

Comment:
Ive had more thought about this, since this is so similar to the one in grid and we plan to add that to @westpac/core we should only have one and just use that version to avoid confusion and be consistent.

I've played around with it and we can refactor the responsive styling options to use it and it ends up being more compact :grinning:

Reply:
Great, keen to see it